### PR TITLE
[#2856] Re-add logic for browser detection using UAParser lib

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/2856-ua-detection-regression

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -22,7 +22,7 @@
             "util/xhr", "util/lang", "util/tutorial",
             "text!default-config.json", "text!layouts/ua-warning.html",
             "ui/widget/tooltip", "crashreporter", "core/project",
-            "util/shims"                  // keep this at the end so it doesn't need a spot in the function signature
+            "UAParser/ua-parser", "util/shims"    // keep these at the end so they don't need a spot in the function signature
```
Comment (dseif) [inline src/main.js]: move `"util/shims"` down a line and put the comment above it as it's pretty long

File: src/main.js [hunk 2]
```diff
@@ -22,7 +22,8 @@
             "util/xhr", "util/lang", "util/tutorial",
             "text!default-config.json", "text!layouts/ua-warning.html",
             "ui/widget/tooltip", "crashreporter", "core/project",
-            "util/shims"                  // keep this at the end so it doesn't need a spot in the function signature
+            "UAParser/ua-parser",
```
Comment (humphd) [inline src/main.js]: Put this on the same line as util/shims so we don't remove them or break things later.  I suggest:

``` javascript
"ui/widget/tooltip", "crashreporter", "core/project",

// keep these at the end so they don't need a spot in the function signature
"UAParser/ua-parser", "util/shims"
```

File: src/main.js [hunk 3]
```diff
@@ -46,21 +49,23 @@
...
 
-      var ua = navigator.userAgent,
-          acceptedUA;
+      // ua-parser uses the current browsers UA by default
+      var ua = new UAParser().getResult(),
+          name = ua.browser.name,
+          major = ua.browser.major,
+          acceptedUA = false;
+
       for ( var uaName in ACCEPTED_UA_LIST ) {
```
Comment (humphd) [inline src/main.js:65]: for..in loops should wrap their body in a hasOwnProperty check.

File: src/main.js (hunk 2 above)
Comment (dseif) [inline src/main.js]: Ah that is my fault. I told him to split up that line, sorry.

**Pull request merged**